kafka-client/src/main/java/dev/responsive/kafka/internal/stores/SegmentedOperations.java

```diff
@@ -0,0 +1,306 @@
+/*
+ *
+ *  * Copyright 2023 Responsive Computing, Inc.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package dev.responsive.kafka.internal.stores;
+
+import static dev.responsive.kafka.internal.config.InternalSessionConfigs.loadSessionClients;
+import static dev.responsive.kafka.internal.config.InternalSessionConfigs.loadStoreRegistry;
+import static org.apache.kafka.streams.processor.internals.ProcessorContextUtils.asInternalProcessorContext;
+import static org.apache.kafka.streams.processor.internals.ProcessorContextUtils.changelogFor;
+
+import dev.responsive.kafka.api.config.ResponsiveConfig;
+import dev.responsive.kafka.api.stores.ResponsiveWindowParams;
+import dev.responsive.kafka.internal.db.CassandraTableSpecFactory;
+import dev.responsive.kafka.internal.db.RemoteWindowedTable;
+import dev.responsive.kafka.internal.db.StampedKeySpec;
+import dev.responsive.kafka.internal.db.partitioning.SubPartitioner;
+import dev.responsive.kafka.internal.metrics.ResponsiveRestoreListener;
+import dev.responsive.kafka.internal.utils.Iterators;
+import dev.responsive.kafka.internal.utils.Result;
+import dev.responsive.kafka.internal.utils.SessionClients;
+import dev.responsive.kafka.internal.utils.Stamped;
+import dev.responsive.kafka.internal.utils.TableName;
+import java.util.Collection;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.processor.StateStoreContext;
+import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.WindowStoreIterator;
+
+public class SegmentedOperations implements WindowOperations {
```

Comment:
Shame on me for not having done this for the other `*Operations` classes (we can do this one as well as a follow up) but we should figure out a testing strategy for it.

Reply:
I actually filed a ticket a while ago for building out a unit testing framework for the state store APIs -- I'll think about it once the baseline window store implementation is done